compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -88,11 +91,12 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                     return;
                 } else {
                     item_msg = access_place_desc;
-                    if self.is_upvar_field_projection(access_place.as_ref()).is_some() {
-                        reason = ", as it is not declared as mutable".to_string();
+                    reason = if self.is_upvar_field_projection(access_place.as_ref()).is_some() {
+                        ", as it is not declared as mutable"
                     } else {
                         let name = self.upvars[upvar_index.index()].to_string(self.infcx.tcx);
-                        reason = format!(", as `{name}` is not declared as mutable");
+                        reason_owned = format!(", as `{name}` is not declared as mutable");
+                        &reason_owned
```

Comment:
Probably just

```suggestion
                        &format!(", as `{name}` is not declared as mutable");
```

Reply:
This is not possible as it would have to extend up multiple scopes, which doesn't seem to currently happen.

Reply:
Shall I undo this change, as it seems a little controversial and I don't care to argue for it?